marlowe/security.md

```diff
@@ -0,0 +1,112 @@
+
+One can compute the script hash of the Marlowe validator from first principles by compiling the validator and computing its hash, assume that one trusts [the Marlowe script source code](src/Language/Marlowe/Scripts.hs) in this repository. Here is a recipe:
+```bash
+nix run 'github:input-output-hk/marlowe-cardano#marlowe-cli' -- contract validator --mainnet --out-file /dev/null --print-hash
+```
+```console
+addr1w94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvq8evxaf
+Validator hash: "6a9391d6aa51af28dd876ebb5565b69d1e83e5ac7861506bd29b56b0"
+```
+Here `addr1w94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvq8evxaf` is the address of the unstaked Marlowe validator and  `6a9391d6aa51af28dd876ebb5565b69d1e83e5ac7861506bd29b56b0` is the well-known script hash of the Marlowe V1 validator.
+
+A staking credential may be associated with the Marlowe validator so that any Ada residing in Marlowe contracts there accrues staking rewards. Thus, the same Marlowe validator might have many different addresses.  For example, here are several addresses associated with the same validator hash:
+```console
+$ nix run 'github:input-output-hk/marlowe-cardano#marlowe-cli' -- contract address --mainnet
+addr1w94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvq8evxaf
+
+$ nix run 'github:input-output-hk/marlowe-cardano#marlowe-cli' -- contract address --mainnet --stake-address stake1u8plvp2a7dythh6yxutd0qlzzkncr3gy5ftvxj02d3etafq4kz6s2
+addr1z94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvxr7cz4mu6gh005gdck67p7y9d8s8zsfgjkcdy75mrjh6jqe7ks37
+
+$ nix run 'github:input-output-hk/marlowe-cardano#marlowe-cli' -- contract address --mainnet --stake-address stake1u8f6d2307veldu24l0u8gck9yp2f0mvxjqtja9ras2wzacss7utvx
+addr1z94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvxn564zluen7mc4t7lcw33v2gz5jlkcdyqh9628mq5u9m3qfuhp8s
+```
+Representing an address as base 16 bytes reveals the payment script hash, starting at the second byte:
+```console
+$ cardano-cli address info --address addr1w94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvq8evxaf | jq -r .base16
+716a9391d6aa51af28dd876ebb5565b69d1e83e5ac7861506bd29b56b0
+
+$ cardano-cli address info --address addr1z94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvxr7cz4mu6gh005gdck67p7y9d8s8zsfgjkcdy75mrjh6jqe7ks37 | jq -r .base16
+116a9391d6aa51af28dd876ebb5565b69d1e83e5ac7861506bd29b56b0c3f6055df348bbdf443716d783e215a781c504a256c349ea6c72bea4
+
+$ cardano-cli address info --address addr1z94f8ywk4fg672xasahtk4t9k6w3aql943uxz5rt62d4dvxn564zluen7mc4t7lcw33v2gz5jlkcdyqh9628mq5u9m3qfuhp8s | jq -r .base16
+116a9391d6aa51af28dd876ebb5565b69d1e83e5ac7861506bd29b56b0d3a6aa2ff333f6f155fbf87462c5205497ed8690172e947d829c2ee2
+```
+
+#### What is the current contract and its state?
+
+The pre-transaction state of the contract is defined in the Plutus `Datum` associated with the UTxO being spent from the Marlowe script address, and this datum must be provided in the transaction.  This datum contains . . .
```

Comment:
Suggest putting a colon after contains instead of the three dots -->> 

This datum contains:

Reply:
Fixed in Fixed in 8a6b693d0d03bd91478b2bc4666c268edefdbd7d..